Update sample app and gloo-gateway refs (#51)
kgateway-dev/kgateway.dev: content/docs/operations/sample-app.md

## Before you begin

Set up {{< reuse "docs/snippets/product-name.md" >}} by following the [Quick start](/docs/quickstart/) or [Installation](/docs/operations/install/) guides.

## Deploy a sample app {#deploy-app}

The following configuration file creates the httpbin app. To review the source file, see [the kgateway project](https://github.com/kgateway-dev/kgateway/blob/main/examples/httpbin.yaml).

1. Create the httpbin app.

   ```shell
   kubectl apply -f https://raw.githubusercontent.com/kgateway-dev/kgateway/refs/heads/main/examples/httpbin.yaml
   ```

   Example output:
   
   ```txt
   namespace/httpbin created
   serviceaccount/httpbin created
   service/httpbin created
   deployment.apps/httpbin created
   ```

2. Verify that the httpbin app is running.
   
   ```sh
   kubectl -n httpbin get pods
   ```

   Example output: 
   
   ```txt
   NAME                      READY   STATUS    RESTARTS   AGE
   httpbin-d57c95548-nz98t   3/3     Running   0          18s
   ```

## Set up an API gateway {#api-gateway}

Create an API gateway with an HTTP listener by using the {{< reuse "docs/snippets/k8s-gateway-api-name.md" >}}.

> 1. Create a gateway resource and configure an HTTP listener. The following gateway can serve HTTP resources from all namespaces.  

Reviewer: ```suggestion
1. Create a Gateway resource and configure an HTTP listener. The following Gateway can serve HTTP resources from all namespaces.  
```

   
   ```yaml
   kubectl apply -f- <<EOF
   kind: Gateway
   apiVersion: gateway.networking.k8s.io/v1
   metadata:
     name: http
     namespace: {{< reuse "docs/snippets/ns-system.md" >}}
   spec:
     gatewayClassName: kgateway
     listeners:
     - protocol: HTTP
       port: 8080
       name: http
       allowedRoutes:
         namespaces:
           from: All
   EOF
   ```

> 2. Verify that the gateway is created successfully. You can also review the external address that is assigned to the gateway. Note that depending on your environment it might take a few minutes for the load balancer service to be assigned an external address. If you are using a local Kind cluster without a load balancer such as `metallb`, you might not have an external address.

Reviewer: ```suggestion
2. Verify that the Gateway is created successfully. You can also review the external address that is assigned to the Gateway. Note that depending on your environment it might take a few minutes for the load balancer service to be assigned an external address. If you are using a local Kind cluster without a load balancer such as `metallb`, you might not have an external address.
```

   
   ```sh
   kubectl get gateway http -n {{< reuse "docs/snippets/ns-system.md" >}}
   ```

   Example output: 
   
   ```txt
   NAME   CLASS      ADDRESS                                  PROGRAMMED   AGE
   http   kgateway   1234567890.us-east-2.elb.amazonaws.com   True         93s
   ```

3. Verify that the gateway proxy pod is running.

   ```sh
   kubectl get po -n {{< reuse "docs/snippets/ns-system.md" >}} -l gateway.networking.k8s.io/gateway-name=http
   ```

   Example output:
   
   ```txt
   NAME                             READY   STATUS    RESTARTS   AGE
   gloo-proxy-http-7dd94b74-k26j6   3/3     Running   0          18s
   ```

   {{< callout type="info" >}}
   Using Kind and getting a `CrashLoopBackOff` error with a `Failed to create temporary file` message in the logs? You might have a multi-arch platform issue on macOS. In your Docker Desktop settings, uncheck **Use Rosetta**, restart Docker, re-create your Kind cluster, and try again.
   {{< /callout >}}

## Expose the app on the gateway {#expose-app}

> Now that you have an app and a gateway, you can create a route to access the app.

Reviewer: ```suggestion
Now that you have an app and a gateway proxy, you can create a route to access the app.
```

> 1. Create an HTTPRoute resource to expose the httpbin app on the gateway. The following example exposes the app on the `wwww.example.com` domain. 

Reviewer: ```suggestion
1. Create an HTTPRoute resource to expose the httpbin app on the Gateway. The following example exposes the app on the `wwww.example.com` domain. 
```

   
   ```yaml
   kubectl apply -f- <<EOF
   apiVersion: gateway.networking.k8s.io/v1
   kind: HTTPRoute
   metadata:
     name: httpbin
     namespace: httpbin
     labels:
       example: httpbin-route
   spec:
     parentRefs:
       - name: http
         namespace: {{< reuse "docs/snippets/ns-system.md" >}}
     hostnames:
       - "www.example.com"
     rules:
       - backendRefs:
           - name: httpbin
             port: 8000
   EOF
   ```

   |Setting|Description|
   |--|--|

>    |`spec.parentRefs`|The name and namespace of the gateway resource that serves the route. In this example, you use the HTTP gateway that you created earlier.  |

Reviewer: ```suggestion
   |`spec.parentRefs`|The name and namespace of the Gateway resource that serves the route. In this example, you use the `http` Gateway that you created earlier.  |
```

   |`spec.hostnames`|A list of hostnames that the route is exposed on. In the example, the route is exposed on `www.example.com`. |

>    |`spec.rules.backendRefs`| The Kubernetes service that serves the incoming request. In this example, requests to `www.example.com` are forwarded to the httpbin app on port 9000. Note that you must create the HTTP route in the same namespace as the service that serves that route. To create the HTTP route resource in a different namespace, you must create a ReferenceGrant resource to allow the HTTP route to forward requests to a service in a different namespace. For more information, see the [Kubernetes API Gateway documentation](https://gateway-api.sigs.k8s.io/api-types/referencegrant/). |

Reviewer: ```suggestion
   |`spec.rules.backendRefs`| The Kubernetes service that serves the incoming request. In this example, requests to `www.example.com` are forwarded to the httpbin app on port 9000. Note that you must create the HTTPRoute in the same namespace as the service that serves that route. To create the HTTPRoute resource in a different namespace, you must create a ReferenceGrant resource to allow the HTTPRoute to forward requests to a service in a different namespace. For more information, see the [Kubernetes API Gateway documentation](https://gateway-api.sigs.k8s.io/api-types/referencegrant/). |
```

2. Verify that the HTTPRoute is applied successfully. 
   
   ```sh
   kubectl get -n httpbin httproute/httpbin -o yaml
   ```

   Example output: Note the status of the HTTPRoute resource. Check for `Accepted` and `ResolvedRefs` messages. The `parentRef` refers to the Gateway that that HTTPRoute is exposed on.

   ```yaml
   status:
     parents:
     - conditions:
       - lastTransitionTime: "2025-02-13T18:41:06Z"
         message: ""
         observedGeneration: 1
         reason: Accepted
         status: "True"
         type: Accepted
       - lastTransitionTime: "2025-02-13T18:41:06Z"
         message: ""
         observedGeneration: 1
         reason: ResolvedRefs
         status: "True"
         type: ResolvedRefs
       controllerName: kgateway.dev/kgateway
       parentRef:
         group: gateway.networking.k8s.io
         kind: Gateway
         name: http
         namespace: kgateway-system
   ```

## Send a request {#send-request}

> Now that your httpbin app is running and you verified that the gateway resources are created, you can send a request to the app. The steps vary depending on your load balancer setup.

Reviewer: ```suggestion
Now that your httpbin app is running and exposed on the gateway proxy, you can send a request to the app. The steps vary depending on your load balancer setup.
```

{{< tabs items="Cloud Provider LoadBalancer,Port-forward for local testing" >}}
{{% tab %}}

> 1. Get the external address of the gateway and save it in an environment variable.

Reviewer: ```suggestion
1. Get the external address of the gateway proxy and save it in an environment variable.
```

   
   ```sh

>    export INGRESS_GW_ADDRESS=$(kubectl get svc -n kgateway-system gloo-proxy-http -o=jsonpath="{.status.loadBalancer.ingress[0]['hostname','ip']}")

Reviewer: ```suggestion
   export INGRESS_GW_ADDRESS=$(kubectl get svc -n kgateway-system kgateway-proxy-http -o=jsonpath="{.status.loadBalancer.ingress[0]['hostname','ip']}")
```

Author: I think it's still called `gloo-proxy` unfortunately, at least for now: https://solo-io-corp.slack.com/archives/C08AAHU0RC0/p1739463892194579?thread_ts=1739303782.523689&cid=C08AAHU0RC0

       ],
       "Host": [
         "www.example.com"
       ],
       "User-Agent": [
         "curl/8.7.1"
       ],
       "X-Envoy-Expected-Rq-Timeout-Ms": [
         "15000"
       ],
       "X-Forwarded-Proto": [
         "http"
       ],
       "X-Request-Id": [
         "26be0bcd-d941-48f4-ac3b-d5ac288ac46f"
       ]
     }
   }
   ```
{{% /tab %}}
{{< /tabs >}}

## Next steps

Now that you have {{< reuse "docs/snippets/product-name.md" >}} set up and running, check out the following guides to expand your API gateway capabilities.

- Add routing capabilities to your httpbin route by using the [Traffic management](/docs/traffic-management) guides. 
- Explore ways to make your routes more resilient by using the [Resiliency](/docs/resiliency) guides. 
- Secure your routes with external authentication and rate limiting policies by using the [Security](/docs/security) guides.

## Cleanup

{{< reuse "docs/snippets/cleanup.md" >}}

1. Delete the httpbin app.

   ```sh
   kubectl delete -f https://raw.githubusercontent.com/kgateway-dev/kgateway/refs/heads/main/examples/httpbin.yaml
   ```

2. Delete the HTTPRoute.

   ```sh
   kubectl delete httproute httpbin -n httpbin
   ```

> 3. Delete the gateway.

Reviewer: ```suggestion
3. Delete the Gateway.
```
